specs/agents/tracing-instrumentation-azure.md

```diff
@@ -0,0 +1,420 @@
+
+Rules derived from the [Queue service REST API reference](https://docs.microsoft.com/en-us/rest/api/storageservices/queue-service-rest-api)
+
+| URL | HTTP verb | HTTP headers | HTTP query string | Resulting Operation Name |
+| --- | --------- | ---------- | ------------------- | ------------------------ |
+| | DELETE    |            |                     | DELETE                   |
+| ends with /messages | DELETE    |            |                     | CLEAR                   |
+| | DELETE    |            | `popreceipt=<value>`| DELETE                   |
+| | GET       |            | `comp=list`         | LISTQUEUES               |
+| | GET       |            | `comp=properties`   | GETPROPERTIES            |
+| | GET       |            | `comp=stats`        | STATS                    |
+| | GET       |            | `comp=metadata`     | GETMETADATA              |
+| | GET       |            | `comp=acl`          | GETACL                   |
+| | GET       |            |                     | RECEIVE                  |
+| | GET       |            | `peekonly=true`     | PEEK                     |
+| | HEAD      |            | `comp=metadata`     | GETMETADATA              |
+| | HEAD      |            | `comp=acl`          | GETACL                   |
+| | OPTIONS   |            |                     | PREFLIGHT                |
+| | POST      |            |                     | SEND                     |
+| | PUT       |            | `comp=acl`          | SETACL                   |
+| | PUT       |            | `comp=properties`   | SETPROPERTIES            |
+| | PUT       |            |                     | CREATE                   |
+| | PUT       |            | `comp=metadata`     | SETMETADATA              |
+| | PUT       |            | `popreceipt=<value>`| UPDATE                   |
+
+### Table storage
+
+The Table service offers non-relational schema-less structured storage in the 
+form of tables. It is a popular service due to its cost-to-capability, though
+messaging in recent years from Microsoft has deprecated Table storage and encouraged
+the use of CosmosDB and its Table storage compatible API.
+
+Tables store data as a collection of entities, where an entity is a collection of properties.
+Entities are similar to rows and properties are similar to columns.
+
+`<ResourceName>` is the name of the resource in the form of the table, or the table and partition key and row key of an entity. For example,
+
+- tablename
+- tablename(PartitionKey='partitionkey', RowKey='rowkey')
+
+`<Storage Account Name>` is the leftmost subdomain of the domain of an Azure storage account endpoint.
+
+| APM field | Required? | Format | Notes | Example |
+| --------- | --------- | ------ | ----- | ------- |
+| `span.name` | yes | `AzureTable <OperationName> <ResourceName>` | Pascal case Operation name | `AzureTable Insert tablename` |

```

Comment:
@felixbarny thoughts on excluding the `<ResourceName>` from the span name? If it were to be removed, is there another appropriate field to capture it?